src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c

```diff
@@ -21,30 +43,242 @@ const ActorInit Obj_Raillift_InitVars = {
+    if (!func_800CAF70(&this->dyna)) {
+        ObjRaillift_UpdatePosition(this, this->curPoint);
+        func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->actionFunc = ObjRaillift_Move;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19BA8.asm")
+void ObjRaillift_Wait(ObjRaillift* this, GlobalContext* globalCtx) {
+    this->waitTimer--;
+    if (this->waitTimer <= 0) {
+        this->actionFunc = ObjRaillift_Move;
+        this->dyna.actor.speedXZ = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19E84.asm")
+void ObjRaillift_Idle(ObjRaillift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, OBJRAILLIFT_GET_FLAG(&this->dyna.actor))) {
+        this->dyna.actor.speedXZ = 0.0f;
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+        this->actionFunc = ObjRaillift_StartCutscene;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19EE0.asm")
+void ObjRaillift_StartCutscene(ObjRaillift* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->cutsceneTimer = 50;
+        this->actionFunc = ObjRaillift_Move;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19F18.asm")
+void ObjRaillift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRaillift* this = THIS;
+    f32 target;
+    f32 step;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19F78.asm")
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->dyna.actor, 10.0f);
+    if (this->cutsceneTimer > 0) {
+        this->cutsceneTimer--;
+        if (this->cutsceneTimer == 0) {
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+    if (OBJRAILLIFT_SHOULD_REACT_TO_WEIGHT(thisx)) {
+        this->hadWeightOn = this->hasWeightOn;
+        if (func_800CAF70(&this->dyna)) {
+            this->hasWeightOn = true;
+        } else {
+            this->hasWeightOn = false;
+        }
+        if ((this->hasWeightOn != this->hadWeightOn) && (this->unk180 < 1.0f)) {
+            this->unk184 = -0x8000;
+            this->unk180 = 6.0f;
+        }
+        this->unk184 += 0xCE4;
+        Math_StepToF(&this->unk180, 0.0f, 0.12f);
+        step = this->hasWeightOn ? Math_CosS(fabsf(this->unk17C) * 2048.0f) + 0.02f : Math_SinS(fabsf(this->unk17C) * 2048.0f) + 0.02f;
+        target = this->hasWeightOn ? -8.0f : 0.0f;
+        Math_StepToF(&this->unk17C, target, step);
+        this->dyna.actor.shape.yOffset = ((Math_SinS(this->unk184) * this->unk180) + this->unk17C) * 10.0f;
+        dummy_label_653499:;
```

Comment:
```suggestion
        dummy:;
```
I don't think there's a convention for this, so it's up to you. But I don't think there's a need for all the random characters that mips2c spits out once it's matching